components/brave_wallet/browser/bitcoin/bitcoin_rpc.cc

```diff
@@ -169,8 +169,7 @@ std::string EndpointHost(const GURL& request_url) {
 
 bool ShouldThrottleEndpoint(const std::string& endpoint_host) {
   // Don't throttle requests if host matches brave proxy.
-  return EndpointHost(GURL(brave_wallet::kBitcoinMainnetRpcEndpoint)) !=
-         endpoint_host;
+  return !brave_wallet::IsEndpointUsingBraveWalletProxy(endpoint_host);
 }
```

Comment:
Let's switch EndpointHost to return URL:
```
GURL EndpointHost(const GURL& request_url) {
  DCHECK(request_url.is_valid());
  return request_url.GetWithEmptyPath();
}
```
and make `ShouldThrottleEndpoint` like
```
bool ShouldThrottleEndpoint(const GURL& endpoint_host) {
  // Don't throttle requests if host matches brave proxy.
  return !brave_wallet::IsEndpointUsingBraveWalletProxy(endpoint_host);
}
```

Reply:
https://github.com/brave/brave-core/pull/24045/commits/b1587bf3552918f346bf44ca2e9f78b35304dfa0